src/partition/ReceivedPartition.cpp

```diff
@@ -89,11 +89,20 @@ void ReceivedPartition::compute()
 {
   PRECICE_TRACE();
 
+  std::cout << "This partition is provided" << std::endl;
+  prepareBoundingBox();
+
   // handle coupling mode first (i.e. serial participant)
   if (not utils::MasterSlave::isSlave() && not utils::MasterSlave::isMaster()) { //coupling mode
     PRECICE_DEBUG("Handle partition data structures for serial participant");
+    mesh::Mesh filteredMesh("FilteredMesh", _dimensions, _mesh->isFlipNormals(), mesh::Mesh::MESH_ID_UNDEFINED);
+    mesh::filterMesh(filteredMesh, *_mesh, [&](const mesh::Vertex &v) { return _bb.contains(v); });
+    _mesh->clear();
+    _mesh->addMesh(filteredMesh);
+
```

Comment:
This is the "coupling mode" case. Meaning a serial participant. It's an open question whether we should filter in this case or not. We don't have to, but I guess you are right that it is more consistent to also filter then.

Reply:
Open question